### ISSUE TITLE
Release under the same license as EditorConfig Core

### ISSUE THREAD

**Issue by bignose-debian** [editorconfig/editorconfig-core-py#20]
The current Python Software Foundation license is [not suitable for third-party works](https://wiki.python.org/moin/PythonSoftwareFoundationLicenseFaq#How_do_I_use_the_PSF_License.3F); neither is it suitable for contributing code to the Python Software Foundation.

Please change all the license grants to be the same as for the rest of EditorConfig Core; I think that would be [the “BSD 2-clause” license](https://spdx.org/licenses/BSD-2-Clause.html). Then, make a new release to PyPI with that license declared in the metadata.

**Comment by xuhdev** [comment]
I read through your links and have also checked out the license file itself. Yes, I agree with you that this is not appropriate. @treyhunner Since only you and me are significant contributors to this project, would you mind changing the license to Apache v2, or something else you prefer?

**Comment by xuhdev** [comment]
Ping @treyhunner

**Comment by xuhdev** [comment]
Ping again... @treyhunner

**Comment by treyhunner** [comment]
@xuhdev yes I agree that we shouldn't use the PSF license. I didn't know what I was doing when I chose it at the time.

I think BSD 2 clause would be most appropriate for this one. I'll change to that now. I can change to Apache later if you have big objections.

**Comment by xuhdev** [comment]
No problem with me.

**Comment by treyhunner** [comment]
Actually I'm not sure how to do this because `fnmatch.py` and `ini.py` are heavily based on code from Python itself, which is released under the PSF license.

Thoughts @xuhdev?

**Comment by xuhdev** [comment]
@treyhunner I think we can simply mark them as exceptions in the license.

**Comment by bignose-debian** [comment]
On 16-Apr-2018, Trey Hunner wrote:
> Actually I'm not sure how to do this because `fnmatch.py` and
> `ini.py` are heavily based on code from Python itself, which is
> released under the PSF license.

Here's what I suggest for dealing with that:

* Include the text of each license in a separate file. For example,
  ‘LICENSE.BSD’ and ‘LICENSE.PSF-2’; each one contains the verbatim
  text of the specific license conditions. Don't change these files
  from their canonical text; it will make it easier for automated
  systems to recognise them in the code base.

* Include a ‘COPYING’ file that states what constitutes the work, and
  that *grants* specific licenses to the recipient. The work as a
  whole is described by name, and the specific parts that you received
  under different license are also named specifically.

  For example:

      Copyright © NNNN–MMMM Trey Hunner and other contributors.

      ‘editorconfig-core-py’ is free software. You are free to copy,
      modify, and/or redistribute this work under the terms of the BSD
      2-clause license. See the file ‘LICENSE.BSD’ for details.

      Parts of this work – the files ‘fnmatch.py’ and ‘ini.py’ –
      are derived from work with copyright held by the Python Software
      Foundation. You are free to copy, modify, and/or redistribute
      those parts under the terms of the Python Software Foundation
      license. See the file ‘LICENSE.PSF-2’ for details.

So, by adding the files ‘COPYING’, and the two files it references
‘LICENSE.BSD’ and ‘LICENSE.PSF-2’, you make very clear what license
applies to what parts.

This meets the requirements of the copyright license received in
‘fnmatch.py’ and ‘ini.py’, while also specifying the license for the
entire work (and for parts of that work absent the PSF works).

This also allows for easily making changes later; you can add new
license conditions, and update ‘COPYING’ to state what grant of
license in what files

I hope that helps.

-- 
 \        "Nothing in life is so exhilarating as to be shot at without |
  `\                                    result."  -- Winston Churchill |
_o__)                                                                  |
Ben Finney <ben@benfinney.id.au>
